### DISCUSSION TITLE
Avoiding Page Level Routing Conflicts with Slug Routes

### DISCUSSION THREAD

**Discussion by SudharakaP** [opencollective/opencollective#4108] [Ideas]
@opencollective/core : This is with regard to https://github.com/opencollective/opencollective/issues/4102. I think it's better to disable page level routing as we are using a custom server, but disabling that [might have some clashes](https://opencollective-frontend-44ktbnx8z-opencollective.vercel.app/pledges/new) with the [`:slug` rewrite](https://github.com/opencollective/opencollective-frontend/blob/86fc5b3cf7b3c2ee80c3b562096b69b5edb96899/rewrites.js#L316). 

My suggestion is to move the `:slug` route to something like `/collectives/:slug` (so a collective url will be something like `www.opencollective.com/collective/babel`) as NextJS gives special value to first order routes (such as `/discover`, `:slug` etc) because of it's page level routing. So when we just use `:slug` we will have to always be careful to make sure our routes don't clash and stuff. What I am thinking is maybe we could avoid this completely by going for `/collectives/:slug` 🤔 

Let me know your thoughts. 😄 

Related to https://github.com/opencollective/opencollective-frontend/pull/6094

**Comment by Betree** [comment]
Using a prefix like `/collectives/:slug` would have been a better design but at this point, it's impossible to consider such a change. Thousands of collectives are relying on their current URLs and have shared them in countless places. We have to support the current pattern, even if that makes things more complex.

**Reply by SudharakaP** [reply]
@Betree : I understand perfectly. Let me think more on this; maybe there's something we can do that makes some improvement. 🤔  For now we'll keep the page level routing (which you applied for dev previews) as it works. 😄
